chameleon/tasks.py

```diff
@@ -131,6 +135,43 @@ def _migrate_project(region, username=None, charge_code=None, access_token=None)
         glance.image_members.update(image.id, ks_legacy_project.id, 'accepted')
         progress += ((1.0 - init_progress) / num_images)
 
+
+    # Migrate cinder volumes if kvm region
+    if region == 'KVM@TACC':
+        unscoped_session = unscoped_user_session(
+            region, access_token=access_token)
+        scoped_user_session = project_scoped_session(
+            project_id=ks_federated_project.id,
+            unscoped_token=unscoped_session.get_token(),
+            region=region)
+
+        admin_cinder = cinder_client.Client(session=sess)
+        user_cinder = cinder_client.Client(session=scoped_user_session)
+
+        volumes_to_migrate = [v for v in admin_cinder.volumes.list(
+            search_opts={
+                'all_tenants': 1,
+                'project_id': ks_legacy_project.id
+            }
+        )]
+
+        num_volumes = len(volumes_to_migrate) 
+        yield progress, f'There are {num_volumes} volumes to process.'
+
+        for index, volume in enumerate(volumes_to_migrate):
+
+            if volume.status == 'available':
+                transfer = admin_cinder.transfers.create(volume.id)
+                user_cinder.transfers.accept(transfer.id, transfer.auth_key)
```

Comment:
Does it not work to use the admin cinder client for this? That could simplify this.

Reply:
Unfortunately, no. The cinder client only accepts a transfer id and auth key so it will transfer the volume to the authenticating project (i.e. `openstack` rather than user's federated project).